### ISSUE TITLE
[Bug]: Error reloading data when restarting milvus

### ISSUE THREAD

**Issue by JokerRiff** [milvus-io/milvus#31145]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:v2.2.0
- Deployment mode(standalone or cluster):standalone
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): CentOS 7.9
- CPU/Memory:16C/64G
- GPU: 
- Others:
```

### Current Behavior

The Milvus database was restarted, and one collection had 1.5 million pieces of data. After waiting for 3 hours, the loaded collections showed a progress of 22%
![error](https://github.com/milvus-io/milvus/assets/50445057/77adf75a-14da-48b3-854a-4cad53bd6c32)

### Milvus Log

[2024/03/08 08:12:58.999 +00:00] [ERROR] [rootcoord/root_coord.go:974] ["failed to describe collection"] [traceID=52f43e0be94264c3] ["collection name"=] [id=446630436343906315] [msgID=0] [ts=446636724511047681] [error="code: CollectionNotExists, reason: can't find collection: oula_qa"] [stack="github.com/milvus-io/milvus/internal/rootcoord.(*Core).DescribeCollection\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/root_coord.go:974\ngithub.com/milvus-io/milvus/internal/distributed/rootcoord.(*Server).DescribeCollection\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/rootcoord/service.go:347\ngithub.com/milvus-io/milvus/internal/proto/rootcoordpb._RootCoord_DescribeCollection_Handler.func1\n\t/go/src/github.com/milvus-io/milvus/internal/proto/rootcoordpb/root_coord.pb.go:1610\ngithub.com/milvus-io/milvus/internal/util/logutil.UnaryTraceLoggerInterceptor\n\t/go/src/github.com/milvus-io/milvus/internal/util/logutil/grpc_interceptor.go:22\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25\ngithub.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34\ngithub.com/milvus-io/milvus/internal/proto/rootcoordpb._RootCoord_DescribeCollection_Handler\n\t/go/src/github.com/milvus-io/milvus/internal/proto/rootcoordpb/root_coord.pb.go:1612\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\t/go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:1283\ngoogle.golang.org/grpc.(*Server).handleStream\n\t/go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:1620\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\t/go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:922"]
[2024/03/08 08:12:58.999 +00:00] [ERROR] [datanode/channel_meta.go:214] ["failed to init bloom filter"] ["segment ID"=446633180697141255] [error="describe collection 446630436343906315 from rootcoord wrong: code: CollectionNotExists, reason: can't find collection: oula_qa"] [stack="github.com/milvus-io/milvus/internal/datanode.(*ChannelMeta).addSegment\n\t/go/src/github.com/milvus-io/milvus/internal/datanode/channel_meta.go:214\ngithub.com/milvus-io/milvus/internal/datanode.(*dataSyncService).initNodes.func2\n\t/go/src/github.com/milvus-io/milvus/internal/datanode/data_sync_service.go:234\ngithub.com/milvus-io/milvus/internal/util/concurrency.(*Pool).Submit.func1\n\t/go/src/github.com/milvus-io/milvus/internal/util/concurrency/pool.go:46\ngithub.com/panjf2000/ants/v2.(*goWorker).run.func1\n\t/go/pkg/mod/github.com/panjf2000/ants/v2@v2.4.8/worker.go:70"]

**Comment by xiaofan-luan** [comment]
We've fixed some of these issues since 2.2.0 is an ancient version of Milvus(Almost two years) and already EOL.
If it's a production environemnt, you could use https://github.com/milvus-io/birdwatcher to hack the meta and fix the env. (If you still need more help we will need your rootcoord log to understand what is happening)

**Comment by xiaofan-luan** [comment]
if it's still under testing, I would recommend to directly use 2.3.10 because fix those problems are very painful and need professional knowledge

**Comment by yanliang567** [comment]
We just released v2.3.11 yesterday, and I believe it was the most stable and powerful release for now.  @JokerRiff I'd suggest you reinstall the latest release.
/assign @JokerRiff 
/unassign

**Comment by JokerRiff** [comment]
Okay, thank you
